Firestore/Swift/Tests/Integration/PipelineTests.swift

```diff
@@ -17,16 +17,147 @@
+    "awards": ["hugo": true, "nebula": false], // Dictionary literal
+    "nestedField": ["level.1": ["level.2": true]], // Nested dictionary literal
+  ],
+  "book2": [
+    "title": "Pride and Prejudice",
+    "author": "Jane Austen",
+    "genre": "Romance",
+    "published": 1813,
+    "rating": 4.5,
+    "tags": ["classic", "social commentary", "love"],
+    "awards": ["none": true],
+  ],
+  "book3": [
+    "title": "One Hundred Years of Solitude",
+    "author": "Gabriel García Márquez",
+    "genre": "Magical Realism",
+    "published": 1967,
+    "rating": 4.3,
+    "tags": ["family", "history", "fantasy"],
+    "awards": ["nobel": true, "nebula": false],
+  ],
+  "book4": [
+    "title": "The Lord of the Rings",
+    "author": "J.R.R. Tolkien",
+    "genre": "Fantasy",
+    "published": 1954,
+    "rating": 4.7,
+    "tags": ["adventure", "magic", "epic"],
+    "awards": ["hugo": false, "nebula": false],
+  ],
+  "book5": [
+    "title": "The Handmaid's Tale",
+    "author": "Margaret Atwood",
+    "genre": "Dystopian",
+    "published": 1985,
+    "rating": 4.1,
+    "tags": ["feminism", "totalitarianism", "resistance"],
+    "awards": ["arthur c. clarke": true, "booker prize": false],
+  ],
+  "book6": [
+    "title": "Crime and Punishment",
+    "author": "Fyodor Dostoevsky",
+    "genre": "Psychological Thriller",
+    "published": 1866,
+    "rating": 4.3,
+    "tags": ["philosophy", "crime", "redemption"],
+    "awards": ["none": true],
+  ],
+  "book7": [
+    "title": "To Kill a Mockingbird",
+    "author": "Harper Lee",
+    "genre": "Southern Gothic",
+    "published": 1960,
+    "rating": 4.2,
+    "tags": ["racism", "injustice", "coming-of-age"],
+    "awards": ["pulitzer": true],
+  ],
+  "book8": [
+    "title": "1984",
+    "author": "George Orwell",
+    "genre": "Dystopian",
+    "published": 1949,
+    "rating": 4.2,
+    "tags": ["surveillance", "totalitarianism", "propaganda"],
+    "awards": ["prometheus": true],
+  ],
+  "book9": [
+    "title": "The Great Gatsby",
+    "author": "F. Scott Fitzgerald",
+    "genre": "Modernist",
+    "published": 1925,
+    "rating": 4.0,
+    "tags": ["wealth", "american dream", "love"],
+    "awards": ["none": true],
+  ],
+  "book10": [
+    "title": "Dune",
+    "author": "Frank Herbert",
+    "genre": "Science Fiction",
+    "published": 1965,
+    "rating": 4.6,
+    "tags": ["politics", "desert", "ecology"],
+    "awards": ["hugo": true, "nebula": true],
+  ],
+]
+
 @available(iOS 13, tvOS 13, macOS 10.15, macCatalyst 13, watchOS 7, *)
 class PipelineIntegrationTests: FSTIntegrationTestCase {
   func testCount() async throws {
     try await firestore().collection("foo").document("bar").setData(["foo": "bar", "x": 42])
     let snapshot = try await firestore()
       .pipeline()
-      .collection(path: "/foo")
-      .where(eq(field("foo"), constant("bar")))
+      .collection("/foo")
+      .where(Field("foo") == Constant("bar"))
```

Comment:
I thought we would move away from operator overload?

Reply:
Yes, I have changed it in the new commit. However, I still want to keep && , ||, and !. I will discuss with Nick later this week.